compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -464,30 +467,44 @@ fn collate_raw_dylibs(used_libraries: &[NativeLib]) -> Vec<(String, Vec<DllImpor
             } else {
                 format!("{}.dll", name)
             };
-            dylib_table
-                .entry(name)
-                .or_default()
-                .extend(lib.dll_imports.iter().map(|import| import.name));
+            dylib_table.entry(name).or_default().extend(lib.dll_imports.iter().cloned());
+        }
+    }
+
+    // Check for multiple imports with the same name but different calling conventions or
+    // (when relevant) argument list sizes.  Rustc only signals an error for this if the
+    // declarations are at the same scope level; if one shadows the other, we only get a lint
+    // warning.
+    for (library, imports) in &dylib_table {
+        let mut import_table: FxHashMap<Symbol, DllCallingConvention> = FxHashMap::default();
+        for import in imports {
```

Comment:
"multiple definitions" are reported by walking an `FxHashSet`, so the walking order is unpredictable and `.stderr` files in the test suite can be different causing the CI failure.

Perhaps this logic needs to be placed after the sorting code below.

Reply:
Yup, just figured this out.  Working on a fix now; should only take an hour or so, barring interrupts from other stuff.